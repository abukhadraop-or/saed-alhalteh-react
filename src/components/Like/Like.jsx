import PropTypes from 'prop-types';
import React from 'react';
import { Like, LikeBox } from './like.styles';

/**
 * Component showing number of likes for this post.
 *
 * @param {Object} props The props object.
 * @param {Number} props.count Number of likes on the current item.
 *
 * @return {JSX.Element} Like component.
 */
function LikeComponent({ count }) {
  return (
    <LikeBox>
      <Like />
      <span>{count}</span>
    </LikeBox>
  );
}

LikeComponent.propTypes = {
  count: PropTypes.number.isRequired,
};

export default LikeComponent;
