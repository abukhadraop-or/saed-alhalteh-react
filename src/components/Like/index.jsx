import PropTypes from 'prop-types';
import React from 'react';
import { Like, LikeBox } from './like.styles';

/**
 * Like button and likes count for the post.
 *
 * @param {object} props The props object.
 * @param {number} props.count Number of likes on the post.
 *
 * @return {JSX.Element} Like component.
 */
function LikeComponent({ count }) {
  return (
    <LikeBox>
      <Like />
      <span>{count}</span>
    </LikeBox>
  );
}

LikeComponent.defaultProps = {
  count: 0,
};

LikeComponent.propTypes = {
  count: PropTypes.number,
};

export default LikeComponent;
