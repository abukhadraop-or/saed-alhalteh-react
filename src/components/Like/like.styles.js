import { ReactComponent as LikeLogo } from 'assets/Icons/likeLogoSolid.svg';
import styled from 'styled-components';
import { colors } from 'shared/styles';

export const LikeBox = styled.div`
  align-self: center;
  border: 2px solid ${colors.fern};
  border-radius: 5px;
  color: ${colors.fern};
  cursor: pointer;
  margin: 0 0 0 auto;
  padding: 5px 15px;
  &:hover {
    background-color: ${colors.fern};
    color: ${colors.white};
    & svg {
      fill: ${colors.white};
    }
  }
  & span {
    vertical-align: middle;
  }
`;
export const Like = styled(LikeLogo)`
  fill: ${colors.fern};
  margin: 0 5px 0 0;
  vertical-align: middle;
`;
