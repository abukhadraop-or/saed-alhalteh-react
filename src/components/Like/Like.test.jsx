import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Like from '.';

const likeProps = {
  count: 0,
};

describe('<Like /> tests', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<Like count={likeProps.count} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
