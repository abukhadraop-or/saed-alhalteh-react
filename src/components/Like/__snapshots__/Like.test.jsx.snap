// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Like /> tests matches the snapshot 1`] = `
<styled.button>
  <Styled(Component) />
  <span>
    0
  </span>
</styled.button>
`;
